service.go

```diff
@@ -88,6 +88,23 @@ const (
 	optionLaunchdConfig = "LaunchdConfig"
 )
 
+// Status represents service status as an interger value
+type Status int
+
+// Status of service represented as an integer
+const (
+	StatusNotImplemented Status = iota
+	StatusUnknown
+	StatusError
+	StatusStarting
```

Comment:
As a crazy idea, what if we had the following status:

StatusUnknown Status = 0
StatusPending Status = 1

StatusRunning Status = 2
StatusPaused Status = 4
StatusStopped Status = 6

StatusStarting = StatusRunning | StatusPending
StatusPausing = StatusPaused | StatusPending
StatusStopping = StatusStopped | StatusPending

This would let consumers test for unknown/error status with a single check and it would let them know if any status was pending with a single check. Then they could proceed with simpler logic for handling Stopped, Running, Paused.

Thoughts?

Reply:
I can do the following, but I have a few questions. 
- How should i represent the `Continue Pending` state for a windows service?
- If StatusUnknown can represent an error, should an error always be returned?

```golang
// Status represents service status as an byte value
type Status byte

// Status of service represented as an byte
const (
	StatusUnknown   Status = 0
	StatusInstalled        = 1 << iota
	StatusPending
	StatusRunning
	StatusPaused
	StatusStopped
	StatusStarting = StatusRunning | StatusPending
	StatusPausing  = StatusPaused | StatusPending
	StatusStopping = StatusStopped | StatusPending
)
```

Reply:
The signature would be `Status() (Status, error)` I think. The happy path would return something like `(StatusRunning, nil)`, while the error would be `(StatusUnknown, fmt.Errorf("yada"))`. Because in the case of an error, the status is unknown.

I would be inclined to represent `Continue Pending` as `Running | Pending`.

For docs, I would say "If the service is not installed, then StatusUnknown and ErrNotInstalled will be returned." Let's not make the install state part of the status. Let's make it a named error if anything.

Reply:
Is the goal of ORing StatusRunning and StatusPending for StatusStarting to be able to represent both states. The reason I ask, is if someone were to decide to use bitwise equality on Status both states would be true as it is right now

I've added an `Is` function below to demonstrate what I mean.

```golang
// Status represents service status as an byte value
type Status byte

// Status of service represented as an byte
const (
	StatusUnknown Status = 1 << iota
	StatusPending
	StatusRunning
	StatusPaused
	StatusContinue
	StatusStopped
	StatusStarting   = StatusRunning | StatusPending
	StatusPausing    = StatusPaused | StatusPending
	StatusContinuing = StatusContinue | StatusPending
	StatusStopping   = StatusStopped | StatusPending
)

// Is returns true if the Status is the provided Status
func (s Status) Is(s2 Status) bool {
	return s&s2 == s2
}
```

Reply:
Yes, I was thinking you could transfer the state that it is both intending to be running and still carry the pending flag at the same time.

I'm a little bit hesitant to include the paused / continue states at all, because neither this library nor any other service system supports it. For that same reason, I'd be inclined to remove the pending states as well. If someone really wants to know every nuance of windows specific state, they can use a platform dependent code.

What the supported states were:
```
// Status represents service status as an byte value
type Status byte

// Status of service represented as an byte
const (
	StatusUnknown Status = iota // Status is unable to be determined due to an error or it was not installed.
	StatusRunning
	StatusStopped
)
```

This is in line with the other platforms, it is in-line with with what this package supports on windows (pause is not supported). It doesn't support the "pending" status, but that is windows specific... as a user, I'm not sure when I'd actually use that bit of information, but it would add 2x number of states to handle each time, for only one platform. Maybe on windows, set the Paused -> Stopped, and Stopping -> Stopped, and Starting -> Running.

Reply:
You know what, that's not a bad idea. Let's just do that. I didn't want to hide detail, but it does make things simpler.